pyvrp/read.py

```diff
@@ -246,18 +283,29 @@ def read(
         for idx in range(len(depot_idcs), dimension)
     ]
 
+    veh_type2idcs = defaultdict(list)
+    for idx, veh_type in enumerate(zip(*vehicle_data)):
+        veh_type2idcs[veh_type].append(idx)
+
     vehicle_types = [
         VehicleType(
             num_available=len(vehicles),
             capacity=capacity,
             depot=depot_idx,
             max_duration=max_duration,
             max_distance=max_distance,
+            fixed_cost=fixed_cost,
+            unit_distance_cost=unit_distance_cost,
             # A bit hacky, but this csv-like name is really useful to track the
             # actual vehicles that make up this vehicle type.
             name=",".join(map(str, vehicles)),
         )
-        for depot_idx, vehicles in enumerate(depot_vehicle_pairs)
+        for (
+            capacity,
+            depot_idx,
+            fixed_cost,
+            unit_distance_cost,
+        ), vehicles in veh_type2idcs.items()
     ]
```

Comment:
I'd probably just write this as a regular loop. This is so long now that an actual loop might be easier to understand. Something like:
```python
vehicle_types = []
for attributes, vehicles in veh_type2idcs.items():
  capacity, depot_idx, fixed_cost, unit_distance_cost = attributes
  vehicle_types.append(...)
```